server/src/com/cloud/network/element/VirtualRouterElement.java

```diff
@@ -568,7 +570,10 @@ public static String getHAProxyStickinessCapability() {
         capabilities.put(Service.SourceNat, sourceNatCapabilities);
 
         capabilities.put(Service.StaticNat, null);
-        capabilities.put(Service.PortForwarding, null);
+
+        final Map<Capability, String> portForwardingCapabilities = new HashMap<Capability, String>();
+        portForwardingCapabilities.put(Capability.SupportedProtocols, NetUtils.TCP_PROTO + "," + NetUtils.UDP_PROTO);
```

Comment:
Why UDP? HAProxy cannot do UDP loadbalancing.

Reply:
@remibergsma 
The changes here are for PF not for LB. Did I get you correctly ?

Reply:
Never mind